src/Storages/StorageReplicatedMergeTree.cpp

```diff
@@ -6403,7 +6390,14 @@ void StorageReplicatedMergeTree::replacePartitionFrom(
             if (code == Coordination::Error::ZOK)
                 delimiting_block_lock->assumeUnlocked();
             else if (code == Coordination::Error::ZBADVERSION)
+            {
+                /// Cannot retry automatically, because some zookeeper ops were lost on the first attempt. Will retry on DDLWorker-level.
+                if (query_context->getZooKeeperMetadataTransaction())
+                    throw Exception(
+                        "Cannot execute alter, because mutations version was suddenly changed due to concurrent alter",
```

Comment:
Need to update exception message, there is not mutation version in ALTER PARTITION